### PR TITLE
Overall fixes, adds kaitaiStructCompile and .editorconfig

### PR DESCRIPTION
**Pull request by KOLANICH**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+root = true
+
+[*]
+charset=utf-8
+indent_style = space
+indent_size = 4
+insert_final_newline = true
```
Comment (GreyCat) [inline .editorconfig]: That's ironic :)

**Comment by GreyCat** [comment]
Please don't bundle so many different changes into one PR, especially calling it just "fixes".

* Adding editorconfig is ok (well, it would have been if the config was actually conformant to what it describes)
* Adding kaitaiStructCompile would probably need to wait at least until the more or less stable pypi release
* Regeneration is ok, but I don't actually get it why are you doing that, if your point is to actually using kaitaiStructCompile to regenerate these on the fly every time

May be some people more proficient in Python than me would comment on the rest (@arekbulski ?)

**Comment by arekbulski** [comment]
I will review the entire PR, just need some time for it. What was exactly that raised your suspicion?

**Comment by GreyCat** [comment]
Me? I have no idea what `setup.cfg` is, what's its status (i.e. recommended / mandatory / deprecated), I have little idea of how imports should be written nowadays, etc.

**Comment by arekbulski** [comment]
Btw, if you need my advice on a PR, please use "request review" next time, for my convenience.

**Comment by KOLANICH** [comment]
> I have no idea what setup.cfg is
A proposal by setuptools authors. Earlier there was another setup.cfg, setuptools devs extended the spec and added into setuptools an own parser. IMHO it is the right thing because a config should be a config, and it should make standalone tools more secure or reliable since working with setup.cfg doesn't involve executing any untrusted python code.

>what's its status (i.e. recommended / mandatory / deprecated)
unofficial. It's just a feature of a de-facto standard library used to build most of python packages.

>I have little idea of how imports should be written nowadays, etc.

One dot - own package. Every additional dot means going up by one level.

**Comment by KOLANICH** [comment]
>Regeneration is ok, but I don't actually get it why are you doing that, if your point is to actually using kaitaiStructCompile to regenerate these on the fly every time.

Just a fallback - if a one doesn't have compiler he can comment out the lines from setup.py and use the pregenerated files.

File: .editorconfig (hunk 1 above)
Comment (arekbulski) [reply .editorconfig]: Ha ha, let me fix that newline from the editor for you.

File: setup.py [hunk 2]
```diff
@@ -1,22 +1,32 @@
 #!/usr/bin/env python
 # -*- coding: utf8 -*-
-from setuptools import setup, find_packages
+#!/usr/bin/env python3
```
Comment (arekbulski) [inline setup.py]: Why does it have 2 shebangs?

**Comment by KOLANICH** [comment]
%hank you for the review, I have splitted into tbe different PRs even though it makes no sense here.

#3 #5#6 #7 #9

**Comment by arekbulski** [comment]
Then why split it?

**Comment by KOLANICH** [comment]
To demonstrate that it shouldn't be splitted of course. I hope there will be no further issues with that anymore.

**Comment by arekbulski** [comment]
Yeah, you made a point there. For the record, I did not have any issues with the PR having multiple commits (or broad scope), just the shorty title, which I edited so there was no problem. I support your point, it should have not been split.

**Pull request closed without merging**